### DISCUSSION TITLE
How to get ref in a backend class function when no ref object is available?

### DISCUSSION THREAD

**Discussion by BrendanBasson4** [rrousselGit/riverpod#1709] [Q&A]
Is it possible to use providers in a backend class without having to pass widget ref around?

For Instance:

```dart
class backendHelperClass {
    void doSomething()
    {
      // Some logic happens here and now UI needs to change accordingly.

      // How do you access a provider from the backend when you don't have a ref object?
      // Am I missing something in the docs?
      // How would you get a ref object without passing it from a widget ?
      final someProviderNotifier = ref.read(someProvider.notifier);

      // Cause UI to update
      someProviderNotifier.update(params);
    }
}
```
I've looked at #1648 but I didn't find what I was looking for.

I'm quite new to using riverpod so please forgive me if it's an obvious answer.

Note: This class is not directly connected to the frontend so passing widget ref would be quite tedious.

**Comment by rrousselGit** [comment]
That class has to be connected with riverpod one way or another.

Your class most likely should be created within a provider. Then, it's have a `Ref` from the provider, which could be passed to the constructor of your class.
